### PR TITLE
Update to Reason 3

### PR DESCRIPTION
**Pull request by jchavarri**
What the title says.

@chenglou The most relevant change is the dependency on reason to be installed globally. Should a note be added to the README?

### CODE REVIEW

File: jeason.sh [hunk 1]
```diff
@@ -23,4 +23,4 @@ SCRIPTDIR="$( cd -P "$( dirname "$THISSCRIPT" )" && pwd )"
 
 SOURCE=$(pwd)/$1
 
-cd $SCRIPTDIR; ./_build/main.native $SOURCE | ./node_modules/reason/refmt_impl.native -use-stdin true -parse binary -print re -is-interface-pp false
+cd $SCRIPTDIR; ./_build/main.native $SOURCE | refmt --parse binary --print re
```
Comment (jchavarri) [inline jeason.sh:26]: I removed `-is-interface-pp` here, seems to not be available and it was false, so...

Comment (chenglou) [reply jeason.sh:26]: Yeah that's an old flag

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "dependencies": {
     "dependency-env": "git+https://github.com/npm-ml/dependency-env.git",
     "nopam": "git+https://github.com/yunxing/nopam.git",
-    "reason": "https://github.com/facebook/reason.git#7cc89888add262a5699cf74802d112f08a4b86a8"
+    "reason": "https://github.com/facebook/reason.git#595db4c5a08e4f5dbe9090152f95720155110a86"
```
Comment (chenglou) [inline package.json]: The local refmt isn't used anymore right?

Comment (jchavarri) [reply package.json]: Oh right, i thought Reason was used for something else. Fixed in https://github.com/chenglou/jeason/pull/11/commits/29da8e1ef45e507369b7b4e77e181c0aa6663c30.

**Pull request merged**